src/compiler/checker.ts

```diff
@@ -29973,8 +29973,11 @@ namespace ts {
         function checkAliasSymbol(node: ImportEqualsDeclaration | ImportClause | NamespaceImport | ImportSpecifier | ExportSpecifier) {
             const symbol = getSymbolOfNode(node);
             const target = resolveAlias(symbol);
-            if (target !== unknownSymbol) {
-                // For external modules symbol represent local symbol for an alias.
+
+            const shouldSkipWithJSRequireTargets = !isInJSFile(node) && moduleKind !== ModuleKind.ES2015;

```

Comment:
I find an ad-hoc change like this suspect (despite our JS support being littered with them). You [said](https://github.com/microsoft/TypeScript/issues/31312#issuecomment-516441263) the differences were caused by the symbol being bound with different flags in JS vs in TS - would that not imply that the appropriate fix lies in adjusting what flags the symbol is bound with, or using the symbol flags that are already different to change the behavior (and not walking up the node tree to find the context file)?

Reply:
OK makes sense, I've changed it - I'm a *little* worried now that before I felt like I definitely got what was going on previously, and now I partially get the change.

Reply:
The only flag you should need to check here is `Assignment` - `Module` comes from the thing being a module, and `Alias` (`AliasExcludes === Alias`, so, y'know, if you see `AliasExcludes` in the debug otuput it means `Alias`) from the thing being an import. Also, you should be able to check for that _instead_ of `isInJSFile(node) && moduleKind !== ModuleKind.ES2015`, not in addition to.

Reply:
Ergh, I'm so sure I tried that earlier and got a tonne of fails. Thanks yeah, that's exactly what it needed.

Reply:
Yeah, the `Assignment` symbol flag is used to mean "allow unusual symbol merges, don't issue an error, deal with the result", since what you wanna suppress here is an error, checking just for that flag means sense.